### PR TITLE
Add a way to set the default value when adding a key to json

### PR DESCRIPTION
**Pull request by cainaleaouk**
The current behaviour when adding a new key to JSON is to set its value to null. I needed it to be empty string, and I thought other people could benefit to setting the default value to something else. 
For instance if you have a JSON with booleans, you could default new entries to true or false.
The code added is little intrusive to the whole system.
I wasn't sure what testing I could add for it, and I'm happy to make one if given directions.
Also the name of the prop `defaultValue` might be too 'wide' for what it does, any naming suggestions appreciated.

### CODE REVIEW

**Comment by mac-s-g** [comment]
@cainaleaouk I think this is a great idea!  I'll take a closer look as soon as I have some time. 

If this is added, behavior should be described in the props table in the README as well.

I would also like a test to be added to make sure it doesn't break in the future.

**Comment by cainaleaouk** [comment]
@mac-s-g Hi Mac, I'm glad you see the value in it. I will update the README. Could you please direct me in what sort of testing you are thinking? I tried coming up with a test but I'm not 100% what to test. 
Should it be added to `AddKeyRequest`? It is the component that uses the prop, but it's not the component that renders the atual input field. Should it be just a prop check or a render check? I'm keen to add them if you could give me some directions on what test you expect :)

File: dev-server/src/index.js [hunk 1]
```diff
@@ -50,6 +50,7 @@ ReactDom.render(
                 }
                 return false
             }}
+            defaultValue=''
```
Comment (franklixuefei) [inline dev-server/src/index.js]: Use double quote for string literals directly assigned to props.

File: index.d.ts [hunk 2]
```diff
@@ -129,6 +129,12 @@ export interface ReactJsonViewProps {
    * Default: false
    */
   sortKeys?: boolean;
+  /**
+   * Set to a value to be used as defaultValue when adding new key to json
+   *
+   * Default: null
+   */
+  defaultValue?: string | number | boolean | null | undefined | string[] | number[] | boolean[] | {}[] | {};
```
Comment (franklixuefei) [inline index.d.ts]: remove `undefined`. `defaultValue?` means optional already.
I suggest you create a new type like 
```ts
type TypeDefaultValue = string | number | boolean | object;
```
and then, you can define the type for `defaultValue` like
```ts
defaultValue?: TypeDefaultValue | TypeDefaultValue[] | null;
```

Please note that the use of `{}` is discouraged. See the following example:

```ts
type t = {};
const c:t = 1; // no error
```
while
```ts
type t = object;
const c:t = 1; // error
```

**Comment by mac-s-g** [comment]
@cainaleaouk this is absolutely awesome!  thank you for putting this together.

I agree with you about the test.  
You'd probably want a test to 
 1. mount the component with a `defaultValue` prop
 2. trigger `submit()`
 3. catch the event that gets emitted 
 4. verify the emitted value matches your expectation

...but that's a little tricky to write and there's not really precedent for it with existing tests.  
so let's skip the test this time.  when I get some time I'll think through how I can test components and resulting events.  that would be valuable for other components as well.

this is a new feature, so there should be a bump to the minor version from 1.18.3 to 1.19.0 associated with the PR.

if you make that change, I'll merge and publish right away.

thanks again for the contribution!

**Comment by mac-s-g** [comment]
love it.  i'll merge and publish.

**Comment by mac-s-g** [comment]
published [+ react-json-view@1.19.0](https://www.npmjs.com/package/react-json-view)

**Pull request merged**